hypervisor/arch/x86/guest/vm.c

```diff
@@ -790,6 +792,9 @@ int32_t shutdown_vm(struct acrn_vm *vm)
 	/* after guest_flags not used, then clear it */
 	vm_config = get_vm_config(vm->vm_id);
 	vm_config->guest_flags &= ~DM_OWNED_GUEST_FLAG_MASK;
+	if ((vm_config->guest_flags & GUEST_FLAG_DYN_VM_CFG) != 0UL) {
+		memset(vm_config, 0U, sizeof(struct acrn_vm_config));
+	}
```

Comment:
Can we clear the entire vm_config? I am not sure.

Reply:
here not clear entire vm_config because we have get the specific vm_config according to the vmid,
vm_config = get_vm_config(vm->vm_id);

Reply:
I didn't catch. You changed name, then you clear the name field, why you reset the entire acrn_vm_config?

Reply:
If the VM name passed from acrn-dm matches one of pre-configured
VM configurations, the corresponding VM will be launched.
If there is no matching found, hypervisor will try to allocate one unused
VM configuration slot for this VM with given VM name and get it run,
in this case, it has set these fields (name guest_flags cpu_affinity load_order) to vm_config, then here clear entire acrn_vm_config.